rust-grammar-dpdfa/grammar.rs

```diff
@@ -35,10 +41,110 @@ pub fn item() {
 
         if peek(Fn) {
             fn_item();
+        } else if peek(Struct) {
+            struct_item();
         } else {
             error();
         }
     }
+
+    if peek() {
+        item();
+    }
+}
+
+fn struct_item() {
+    if peek(Pub) {
+        vis();
+    }
```

Comment:
May be easier to create a `fn vis_opt()` in order to wrap away all these `if peek { vis() }`